sdks/java/core/src/main/java/org/apache/beam/sdk/schemas/Schema.java

```diff
@@ -182,6 +184,33 @@ public boolean equals(Object o) {
         && Objects.equals(getFields(), other.getFields());
   }
 
+  /** Returns true if two Schemas have the same fields, but possibly in different orders. */
+  public boolean equivalent(Schema other) {
+    List<Field> otherFields =
+        other
+            .getFields()
+            .stream()
+            .sorted(Comparator.comparing(Field::getName))
+            .collect(Collectors.toList());
+    List<Field> actualFields =
+        getFields()
+            .stream()
+            .sorted(Comparator.comparing(Field::getName))
+            .collect(Collectors.toList());
+    if (otherFields.size() != actualFields.size()) {
```

Comment:
I'm pretty sure I've seen this function before on the whiteboard in an interview. It would be good to move this check before the sorts.

Reply:
Good call, done